### DISCUSSION TITLE
defaultChecked not working in my custom component v7

### DISCUSSION THREAD

**Discussion by c0d3x** [react-hook-form/react-hook-form#5228] [Q&A] [answered]
I am having a hard time getting the checkbox checked value from defaultValues when it is updated by state.
I used to have the defaultChecked in useForm that used the updated state, but that one is removed from v7.. any clues?
The checkbox uses the event.target.checked vs the event.target.value.. so is there implemented logic for this?

**Comment by c0d3x** [comment]
Here is some code I worked on trying to get checkbox to work.
https://codesandbox.io/s/react-hook-form-checkbox-gy2my?file=/src/index.tsx

**Comment by thanh-nguyen-95** [comment, marked as answer]
This works without changing the `Checkbox` component. Default state of the checkbox follows the `terms` state.

```jsx
<Controller
  control={control}
  name="confirmTerms"
  defaultValue={terms}
  render={({ field }) => (
    <Checkbox
      {...field}
      id="confirm-terms"
      label="Please Confirm Terms"
      errorText={errors?.confirmTerms?.message}
      onChange={(e) => {
        const { target: { checked }} = e
        
        setTerms(checked);
        field.onChange(checked);
      }}
    />
  )}
/>
```

**Reply by c0d3x** [reply]
I am not using the Controller in my form, seems it worked when I prop spread {...rest}

**Reply by thanh-nguyen-95** [reply]
My apology. It's a habit of mine to use `Controller` when dealing with custom components.

I think the important part here is `onChange`. You need to change both your state and `react-hook-form`'s state.

If no further problems arise, please mark this as answered.
